types/azure-kusto-data/index.d.ts

```diff
@@ -0,0 +1,75 @@
+// Type definitions for azure-kusto-data 0.2
+// Project: https://github.com/Azure/azure-kusto-node/tree/master/azure-kusto-data
+// Definitions by: Armando Aguirre <https://github.com/armanio123>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.1
+
+export class Client {
+    constructor(kcsb: string | KustoConnectionStringBuilder);
+    connectionString: KustoConnectionStringBuilder;
+    cluster: string;
+    endpoints: { mgmt: string, query: string };
+    aadHelper: any;
+    execute(db: any, query: any, callback: (err: Error, results: Client.KustoResponseDataSetV2<any>) => void, properties?: any): any;
+    executeMgmt(db: any, query: any, callback: (err: Error, results: Client.KustoResponseDataSetV2<any>) => void, properties: any): any;
+    executeQuery(db: any, query: any, callback: (err: Error, results: Client.KustoResponseDataSetV2<any>) => void, properties: any): any;
+}
+
+export class ClientRequestProperties {
+    clearOptions(): void;
+    getOption(name: any, defaultValue: any): any;
+    getTimeout(): any;
+    setOption(name: any, value: any): void;
+    setTimeout(timeoutMillis: any): void;
+    toJson(): any;
+}
+
+export class KustoConnectionStringBuilder {
+    constructor(connectionString: string);
+    dataSource: string;
```

Comment:
this class is to be used via the static methods, can we not expose the internal properties so we don't confuse users?

Reply:
I can remove the properties to avoid having users trying them. But I think nonetheless, the properties are accessible and any user reading through the code will be able to find them and use them.

Let me know if you still want me to remove it.

Reply:
@danield137 let me know how you feel about this otherwise I'll merge it tomorrow.

Reply:
I don't want to hold this up, it's better to be out there. 
If any misunderstands happen, we will fix it in the future. 
Go ahead and merge.